src/libstd/io/mod.rs

```diff
@@ -90,6 +91,16 @@ fn append_to_string<F>(buf: &mut String, f: F) -> Result<usize>
     }
 }
 
+/// Grow `buf` by `additional` bytes after current length, and fill with zeros.
+fn grow_zerofill(buf: &mut Vec<u8>, additional: usize) {
+    let len = buf.len();
+    buf.reserve(additional);
+    unsafe {
+        buf.set_len(len + additional);
+        buf[len..].set_memory(0);
+    }
+}
```

Comment:
Maybe this should be a method on `Vec<u8>` or even `Vec<T> where T: Copy` in general (with element supplied).